### PR TITLE
Selector filter

### PR DESCRIPTION
**Pull request by ubitux**

### CODE REVIEW

File: tests/filter.py [hunk 1]
```diff
@@ -132,6 +134,32 @@ def filter_saturation(_):
...
+    )
+
+
+@test_cuepoints(points=_CUEPOINTS, keyframes=1, tolerance=1)
+@ngl.scene()
+def filter_selector_hue(_):
+    return _base_scene(
+        ngl.FilterSelector(
+            range=(-0.1 * math.tau, 0.45 * math.tau),
+            component="hue",
```
Comment (mbouron) [inline tests/filter.py:156]: Can you also exercise the "chroma" component ?

File: libnopegl/src/node_filters.c [hunk 2]
```diff
@@ -191,6 +207,71 @@ static const struct node_param filtersaturation_params[] = {
 };
 #undef OFFSET
 
+enum {
+    SELECTOR_COMPONENT_LIGHTNESS,
+    SELECTOR_COMPONENT_CHROMA,
+    SELECTOR_COMPONENT_HUE,
```
Comment (mbouron) [inline libnopegl/src/node_filters.c:213]: Can you explicitly set their value since `2` is directly referenced in the shader ?

**Review by mbouron — APPROVED** [review]
Except from my comment, LGTM.

File: tests/filter.py (hunk 1 above)
Comment (ubitux) [reply tests/filter.py:169]: added

File: libnopegl/src/node_filters.c (hunk 2 above)
Comment (ubitux) [reply libnopegl/src/node_filters.c]: done for all 3 enums, with a comment to explain why

**Pull request merged**